poly-commit/src/hyrax/mod.rs

```diff
@@ -503,15 +497,22 @@ where
             // it from the transcript.
             let c: G::ScalarField = sponge.squeeze_field_elements(1)[0];
 
-            // First check
-            let com_z_zd = (Self::pedersen_commit(&vk.com_key, z) + vk.h * z_d).into();
-            if com_z_zd != (t_prime.mul(c) + com_d).into() {
+            // Second check from the paper (figure 6, equation (14))
+            // Moved here for potential early return
```

Comment:
Why is this check called "second" if it is not the second check? Maybe remove "first" and "second" and only refer to the equation number?

Reply:
They are commented as "first check from the paper" and "second check from the paper" (not just as "first" and "second") because that's the order they appear in the paper - which is true regardless of our implementation. It just so happens that it's (potentially) more efficient to make them in the opposite order for early returns. I don't think "first check from the paper" and "second check from the paper" can lead to any confusion, right?

Reply:
Hmm, I see. I thought the equation number would be enough. But true, it is not really confusing.

Reply:
My commit was unsigned, I had to rebase and repush your commit.

Reply:
No problem about the rebasing :) thanks for the fix